### PR TITLE
Ci/163 merge jenkins files

### PR DESCRIPTION
**Pull request by StateOf-Vlado**
Jenkins files were merged
https://github.com/status-im/desktop-qa-automation/issues/163

### CODE REVIEW

File: ci/Jenkinsfile [hunk 1]
```diff
@@ -66,21 +71,36 @@ pipeline {
...
+          /* Creating directory for application build */
+          sh "mkdir -p '${env.WORKSPACE}/tmp/pkg/'"
+          def job_name = "${JOB_NAME}"
+          /* Searching the latest build of the master branch is available only for ci.infra.status.im */
+          if(job_name.endsWith("nightly")){
+            echo "Searching the latest successful build for the master branch."
+            if(params.BUILD != ""){
+              echo "BUILD value was ignored. Please use 'ci.status.in' to test specific builds."
+            }
+            /* Seraching and downloading the latest successful build for the master branch */
```
Comment (Valentina1133) [inline ci/Jenkinsfile:87]: typo in first word

**Review by Valentina1133 — APPROVED** [review]

File: ci/Jenkinsfile [hunk 2]
```diff
@@ -68,18 +71,33 @@ pipeline {
...
+            copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
+          } else {
+            if(params.BUILD == ""){
+              throw new Exception("BUILD is required field.")
+            }
+            withCredentials([usernamePassword(credentialsId: 'status-im-auto', usernameVariable: 'CI_CREDS_USR', passwordVariable: 'CI_CREDS_PSW')]) {
+              fileOperations([
+                fileDownloadOperation(url:"${params.BUILD}", userName:"${env.CI_CREDS_USR}", password:"${env.CI_CREDS_PSW}", targetLocation:"./tmp/pkg/", targetFileName:"StatusIm-Desktop.tar.gz", proxyHost:"", proxyPort:"")
```
Comment (jakubgs) [inline ci/Jenkinsfile:95]: If you insist on supporting file download, it should not use any auth, and it should use publicly available build URLs.

And it should be in a separate stage using `when` clauses.

File: ci/Jenkinsfile [hunk 3]
```diff
@@ -15,10 +15,8 @@ pipeline {
     PYTHONPATH = "${SQUISH_DIR}/lib:${SQUISH_DIR}/lib/python:${PYTHONPATH}"
     LD_LIBRARY_PATH = "${SQUISH_DIR}/lib:${SQUISH_DIR}/python3/lib:${LD_LIBRARY_PATH}"
 
-    TESTRAIL_CREDS = credentials('test-rail-api-devops')
     TESTRAIL_URL = "https://ethstatus.testrail.net"
-
-    CI_CREDS = credentials('status-im-auto')
+    TESTRAIL = credentials('test-rail-api-devops')
```
Comment (jakubgs) [inline ci/Jenkinsfile:19]: please don't set credentials in global `environment` section. Use `withCredentials` in the stage in which it is needed.

File: ci/Jenkinsfile [hunk 4]
```diff
@@ -35,9 +33,14 @@ pipeline {
       tagFilter: '*',
       type: 'PT_BRANCH'
     )
+    booleanParam(
+      name: 'FIND_BUILD',
```
Comment (yakimant) [inline ci/Jenkinsfile]: As I can see `FIND_BUILD` is not used - let's remove it,

File: ci/Jenkinsfile [hunk 5]
```diff
@@ -35,9 +33,14 @@ pipeline {
...
+      description: 'Use the latest successful build from the master branch.',
+      defaultValue: true
+    )
     string(
       name: 'BUILD',
```
Comment (yakimant) [inline ci/Jenkinsfile:47]: Let's rename to `BUILD_URL` for clarity

File: ci/Jenkinsfile (hunk 2 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: Credentials removed

File: ci/Jenkinsfile (hunk 4 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: deleted, we will search build if the BUILD field is empty

File: ci/Jenkinsfile (hunk 3 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: fixed

File: ci/Jenkinsfile [hunk 6]
```diff
@@ -35,8 +28,20 @@ pipeline {
...
+      choices: [
+      'linux',
+      'linux-01',
+      'linux-02',
+      'linux-03',
+      'linux-04',
+      'linux-05',
+      ],
+      description: 'Agent name to run tests on it. '
+    )
```
Comment (jakubgs) [inline ci/Jenkinsfile:42]: What is the point of this?

File: ci/Jenkinsfile [hunk 7]
```diff
@@ -70,16 +79,51 @@ pipeline {
...
+          } else {
+
+            /* Manual run and PRs trigger for status-desktop */
+
+            if(params.BUILD_URL == ""){
+              throw new Exception("BUILD_URL is required field.")
+            } else if(params.BUILD_URL.contains("ci.infra.status.im")) {
+              throw new Exception("Please use 'https://ci.infra.status.im/job/status-desktop/job/manual/' to test specific master/release builds.")
+            }
+            withCredentials([usernamePassword(credentialsId: 'status-im-auto', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]){
```
Comment (jakubgs) [inline ci/Jenkinsfile:118]: Why do we need credentials of any kind at all? Does this really need to be so complex?

Also, why can't we just use a file parameter? Instead of doing all of this silliness:
https://dumbitdude.com/how-to-upload-a-file-in-jenkins-using-file-parameter/

File: ci/Jenkinsfile [hunk 8]
```diff
@@ -70,16 +79,51 @@ pipeline {
...
-        }
-        fileOperations([
-          fileDownloadOperation(url:"${params.BUILD}", userName:"${CI_CREDS_USR}", password:"${CI_CREDS_PSW}", targetLocation:"./tmp", targetFileName:"StatusIm-Desktop.tar.gz", proxyHost:"", proxyPort:"")
-        ])
-        script {
-          sh "tar -zxvf '${env.WORKSPACE}/tmp/StatusIm-Desktop.tar.gz' -C '${env.WORKSPACE}/tmp'"
+          /* Creating directory for application build */
+          sh "mkdir -p '${env.WORKSPACE}/tmp/pkg/'"
+          def job_name = "${JOB_NAME}"
+          if(job_name.endsWith("nightly")){
```
Comment (jakubgs) [inline ci/Jenkinsfile:88]:
```suggestion
          if (JOB_NAME.endsWith("nightly")) {
```

File: ci/Jenkinsfile [hunk 9]
```diff
@@ -70,16 +79,51 @@ pipeline {
...
+
+             /* Nightly regression */
+
+            if(params.BUILD_URL != ""){
```
Comment (jakubgs) [inline ci/Jenkinsfile:92]: You keep eating whitespace man.
```suggestion
            if (params.BUILD_URL != "") {
```

File: ci/Jenkinsfile [hunk 10]
```diff
@@ -70,16 +79,51 @@ pipeline {
...
+              echo "BUILD_URL value was ignored. Please use 'https://ci.infra.status.im/job/status-desktop/job/e2e/job/manual/' to test specific builds."
+            }
+            copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
```
Comment (jakubgs) [inline ci/Jenkinsfile:95]: Long lines are hard to follow.
```suggestion
            copyArtifacts(
              projectName: "status-desktop/systems/linux/x86_64/package/",
              filter: 'pkg/*-x86_64.tar.gz',
              selector: lastWithArtifacts(),
              target: "${env.WORKSPACE}/tmp"
            )
```
The same goes for any long call

Comment (jakubgs) [reply ci/Jenkinsfile:95]: Also, you're already in `WORKSPACE` folder, why do you need to specify it?

File: ci/Jenkinsfile [hunk 11]
```diff
@@ -70,16 +79,51 @@ pipeline {
...
-          sh "tar -zxvf '${env.WORKSPACE}/tmp/StatusIm-Desktop.tar.gz' -C '${env.WORKSPACE}/tmp'"
+          /* Creating directory for application build */
+          sh "mkdir -p '${env.WORKSPACE}/tmp/pkg/'"
+          def job_name = "${JOB_NAME}"
+          if(job_name.endsWith("nightly")){
+
+             /* Nightly regression */
+
+            if(params.BUILD_URL != ""){
+              echo "BUILD_URL value was ignored. Please use 'https://ci.infra.status.im/job/status-desktop/job/e2e/job/manual/' to test specific builds."
```
Comment (jakubgs) [inline ci/Jenkinsfile:93]: By why?

File: ci/Jenkinsfile [hunk 12]
```diff
@@ -92,26 +136,24 @@ pipeline {
...
+            sh "fluxbox &"
+            def cmd = ''
+            if (params.TEST_NAME != "") {
+              cmd = cmd + " -k ${params.TEST_NAME}"
+            }
+            if (params.TEST_SCOPE != "") {
+            cmd = cmd + " -m ${params.TEST_SCOPE}"
+            }
+            sh "cp -f '${env.WORKSPACE}'/configs/_local.py.ci '${env.WORKSPACE}'/configs/_local.py"
+            withCredentials([usernamePassword(credentialsId: 'test-rail-api-devops', usernameVariable: 'TESTRAIL_USR', passwordVariable: 'TESTRAIL_PSW')]){
```
Comment (jakubgs) [inline ci/Jenkinsfile:151]:
```suggestion
            withCredentials([
              usernamePassword(
                credentialsId: 'test-rail-api-devops',
                usernameVariable: 'TESTRAIL_USR',
                passwordVariable: 'TESTRAIL_PSW')
              ]){
```

File: ci/Jenkinsfile [hunk 13]
```diff
@@ -70,16 +79,51 @@ pipeline {
...
+            if(params.BUILD_URL != ""){
+              echo "BUILD_URL value was ignored. Please use 'https://ci.infra.status.im/job/status-desktop/job/e2e/job/manual/' to test specific builds."
+            }
+            copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
+          } else if (job_name.endsWith("manual")){
+
+             /* Manual run and PRs trigger for desktop-qa-automation */
+
+            if(params.BUILD_URL == ""){
+              copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
```
Comment (jakubgs) [inline ci/Jenkinsfile:101]: I really think this should be separate stages. This wall of code is hard to follow. It should be two mutually exclusive stages: `Copy` and `Download`(if we want to support that) with `when` clauses that decide which one will run.

And please stop making lines so long.

File: ci/Jenkinsfile (hunk 6 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile:42]: I'm testing why some of the agents don't have a file dialog during test execution
I want to select an agent to execute tests

File: ci/Jenkinsfile (hunk 7 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: fixed

File: ci/Jenkinsfile (hunk 8 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: fixed

File: ci/Jenkinsfile (hunk 9 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: fixed

File: ci/Jenkinsfile (hunk 10 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: Why are you asking me about it? If you know how to make it in another way just write it here and say that it is better to use it like this. Let's save our time, "men"?

File: ci/Jenkinsfile (hunk 11 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: removed

File: ci/Jenkinsfile (hunk 12 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: Thanks for the idea, fixed

File: ci/Jenkinsfile (hunk 13 above)
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: I removed the copy-paste of methods, so we have only one downloading and one copy

File: ci/Jenkinsfile [hunk 14]
```diff
@@ -68,18 +71,33 @@ pipeline {
...
       steps {
         script {
-          sh "mkdir '${env.WORKSPACE}/tmp/'"
-        }
-        fileOperations([
-          fileDownloadOperation(url:"${params.BUILD}", userName:"${CI_CREDS_USR}", password:"${CI_CREDS_PSW}", targetLocation:"./tmp", targetFileName:"StatusIm-Desktop.tar.gz", proxyHost:"", proxyPort:"")
-        ])
-        script {
-          sh "tar -zxvf '${env.WORKSPACE}/tmp/StatusIm-Desktop.tar.gz' -C '${env.WORKSPACE}/tmp'"
+          /* Creating directory for application build */
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply ci/Jenkinsfile]: fixed

File: ci/Jenkinsfile [hunk 15]
```diff
@@ -92,26 +115,24 @@ pipeline {
...
+            sh "fluxbox &"
+            def cmd = ''
+            if (params.TEST_NAME != "") {
+              cmd = cmd + " -k ${params.TEST_NAME}"
+            }
+            if (params.TEST_SCOPE != "") {
+            cmd = cmd + " -m ${params.TEST_SCOPE}"
+            }
+            sh "cp -f '${env.WORKSPACE}'/configs/_local.py.ci '${env.WORKSPACE}'/configs/_local.py"
+            withCredentials([usernamePassword(credentialsId: 'test-rail-api-devops', usernameVariable: 'TESTRAIL_USR', passwordVariable: 'TESTRAIL_PSW')]) {
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply ci/Jenkinsfile:130]: fixed

File: ci/Jenkinsfile [hunk 16]
```diff
@@ -70,16 +79,30 @@ pipeline {
...
+            copyArtifacts(
+              projectName:  "status-desktop/systems/linux/x86_64/package/",
+              filter:       'pkg/*-x86_64.tar.gz',
+              selector:     lastWithArtifacts(),
+              target:       "./tmp"
+            )
+          }
+          def pkg_path = "${env.WORKSPACE}/${utils.findFile('tmp/pkg/*tar.gz')}"
+          sh "tar -zxvf '${pkg_path}' -C '${env.WORKSPACE}/tmp'"
           env.APP_DIR = "${env.WORKSPACE}/${utils.findFile('tmp/*.AppImage')}"
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply ci/Jenkinsfile:106]: fixed

File: ci/Jenkinsfile [hunk 17]
```diff
@@ -35,8 +28,20 @@ pipeline {
...
     )
+    choice(
+      name: 'AGENT',
+      choices: [
+      'linux',
+      'linux-01',
+      'linux-02',
+      'linux-03',
+      'linux-04',
+      'linux-05',
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply ci/Jenkinsfile:39]: Where?

File: ci/Jenkinsfile [hunk 18]
```diff
@@ -70,16 +79,30 @@ pipeline {
   }
 
   stages {
-    stage('Download') {
+     stage('Download') {
```
[the remark this answers is not in the document]
Comment (StateOf-Vlado) [reply ci/Jenkinsfile:82]: fixed
